packages/@lwc/engine-server/src/__tests__/fixtures/scoped-slots/basic/expected.html

```diff
@@ -2,11 +2,13 @@
   <template shadowroot="open">
     <x-basic-child>
       <div>
-        ‍‍
+        <!---->
+        <!---->
```

Comment:
Do you know why these are turning into 2 comment nodes rather than a single comment node?

Reply:
These were two empty (or zero-width, when serialized) text nodes prior to this PR; it didn't appear to be that way because the text node content was invisible.

There are actually two fragments involved here, one nested inside the other. One fragment corresponds to the slot in the child where the content is inserted. The other fragment corresponds to the `lwc:slot-data` template in the parent. When the VDOM is combined, the originating-from-parent VFragment is nested inside the accepting-content-in-child VFragment, which is how you end up with these two comment-node anchors.